weaviate/README.md

```diff
@@ -10,14 +10,58 @@ Follow the instructions below to install and configure this check for an Agent r
 
 ### Installation
 
-The Weaviate check is included in the [Datadog Agent][2] package.
-No additional installation is needed on your server.
+The Weaviate check is included in the [Datadog Agent][2] package. No additional installation is needed on your server.
 
-### Configuration
+**Note**: This check requires Agent v7.47.0+.
 
-1. Edit the `weaviate.d/conf.yaml` file, in the `conf.d/` folder at the root of your Agent's configuration directory to start collecting your weaviate performance data. See the [sample weaviate.d/conf.yaml][4] for all available configuration options.
+### Configuration
 
-2. [Restart the Agent][5].
+Weaviate can be configured to expose Prometheus-formatted metrics. The Datadog Agent can collect these metrics using the integration described below. Follow the instructions to configure data collection from any or all of the components. For the required configurations, please refer to the [Weaviate documentation][11].
+
+In addition, a small subset of metrics can be collected by communicating with different [API endpoints][12].
+
+**Note**: This check uses [OpenMetrics][12] for metric collection, which requires Python 3.
+
+#### Containerized
+##### Metric collection
+
+Make sure that the Prometheus-formatted metrics are exposed in your Weaviate cluster. You can configure and customize this by following the [Weaviate documentation][11]. In order for the Agent to start collecting metrics, the Weaviate pods need to be annotated. For more information about annotations, refer to the [Autodiscovery Integration Templates][3] for guidance. You can find additional configuration options by reviewing the [sample weaviate.d/conf.yaml][4]
+
+Note: The listed metrics can only be collected if they are available. Some metrics are generated only when certain actions are performed. For example, the object deletion metric will be exposed only when an object is deleted.
+
+The two most important parameters for configuring the Weaviate check are as follows:
+- `openmetrics_endpoint`: This parameter should be set to the location where the Prometheus-formatted metrics are exposed. The default port is 2112, but it can be configured using the PROMETHEUS_MONITORING_PORT [environment variable][10]. In containerized environments, `%%host%%` can be used for host autodetection [3]. 
```

Comment:
```suggestion
- `openmetrics_endpoint`: This parameter should be set to the location where the Prometheus-formatted metrics are exposed. The default port is `2112`, but it can be configured using the `PROMETHEUS_MONITORING_PORT` [environment variable][10]. In containerized environments, `%%host%%` can be used for [host autodetection][3]. 
```